src/UglyToad.PdfPig/Graphics/ContentStreamProcessor.cs

```diff
@@ -315,7 +315,16 @@ public void ShowText(IInputBytes bytes)
                 var transformedPdfBounds = PerformantRectangleTransformer
                     .Transform(renderingMatrix, textMatrix, transformationMatrix, new PdfRectangle(0, 0, boundingBox.Width, 0));
 
-                      
+                if (parsingOptions.ClipPaths)
+                {
+                    var currentClipping = GetCurrentState().CurrentClippingPath;
+
+                    if (!currentClipping?.Contains(transformedGlyphBounds) ?? true)
```

Comment:
No need to call currentState `GetCurrentState()` again here, you can reuse the `currentState` variable from line 246

Reply:
I think for the corner case where the clipping path would be null, we need to render the text (i.e. we don't skip). You can achieve that using something like 
```csharp
currentState.CurrentClippingPath?.Contains(transformedGlyphBounds) == false
```

Reply:
@savasozer no worries at all! I agree with you, using `Intersect()` makes more sense

Reply:
I couldn't find a PdfPath PdfRectangle Intersection testing method. So I made it myself using your Contains method :)

Reply:
@savasozer thanks a lot for that, that's very helpful.

Maybe 1 remark on your function (I'm away so I could not actually test it):
- There is a (very rare) edge case I think where the clipping path is completely inside the letter's rectangle, leading your function to return _false_ instead of _true_. Same if the clipping path is smaller than the letter and partially inside and outside of the letter. In both cases all letters point would be outside but actually overlap. There might be other edge cases but I could only think about this one.

Reply:
Maybe this link would be useful https://stackoverflow.com/questions/38382777/how-do-i-determine-if-two-polygons-intersect-using-clipper#38382810